src/ansible/roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,148 @@
+- name: print keycloak_url
+  debug:
+    msg: "keycloak_url={{ keycloak_url }}"
+
+- name: Download software zipfile
+  get_url:
+    url: "{{ keycloak_url }}"
+    dest: /tmp
+    
+- name: Unzip software
+  unarchive:
+    remote_src: yes
+    src: "/tmp/{{ zip_file }}"
+    dest: /tmp
+
+- name: Check if unzipped directory exists
+  stat:
+    path: "{{ unzip_dir }}"
+  register: unzip_dir_chk
+
+- name: debug unzip dir
+  debug:
+    var: unzip_dir_chk.stat
+
+- name: Check if /opt/keycloak dir exists
+  stat:
+    path: /opt/keycloak
+  register: opt_keycloak_chk
+
+- name: debug opt dir
+  debug:
+    var: opt_keycloak_chk.stat
+
+- name: Move software to /opt
+  command: mv {{ unzip_dir }} /opt/keycloak 
+  when: unzip_dir_chk.stat.exists and not opt_keycloak_chk.stat.exists
+
+- name: create Keycloak user/group
+  user:
+    name: keycloak
+    home: /opt/keycloak
+    system: yes
+    create_home: no
+
+- name: change ownership of files in /opt/keycloak
+  file:
+    path: /opt/keycloak
+    owner: keycloak
+    group: keycloak
+    recurse: yes
+
+- name: change ownership of files in /data/certs
+  file:
+    path: /data/certs
+    owner: keycloak
+    group: keycloak
+    recurse: yes
+
+- name: Add CA certificate to keystore
+  shell: |
+    keytool -noprompt -import \
+      -keystore /data/certs/master.keycloak.test.keystore \
+      -file /data/certs/ca.crt \
+      -alias ca.crt \
+      -trustcacerts -storepass {{ service.keycloak.admin_password }}
+  args:
+    creates: /data/certs/master.keycloak.test.keystore
+
+- name: Add Keycloak certificate to keystore
+  shell: |
+    keytool -noprompt -import \
+      -keystore /data/certs/master.keycloak.test.keystore \
+      -file /data/certs/master.keycloak.test.crt \
+      -alias master.keycloak.test.crt \
+      -trustcacerts -storepass {{ service.keycloak.admin_password }}
+
+- name: Copy CA Certificate to local pki anchors
+  copy:
+    src: /data/certs/ca.crt
+    dest: "{{ ca_trust_dir }}"
+    remote_src: yes
+
+- name: Update system ca trusts to pick up new certificate
+  command: "{{ ca_trust_update }}"
+
+- name: create /etc/sysconfig if it doesn't exist
+  file:
+    name: /etc/sysconfig
+    state: directory
```

Comment:
Is this needed?

Reply:
I thought I had added it for a specific reason but, it does not appear to be needed.   Removed.